spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/support/DefaultPublisherFactory.java

```diff
@@ -66,8 +66,8 @@ public DefaultPublisherFactory(GcpProjectIdProvider projectIdProvider,
 		Assert.notNull(executorProvider, "The executor provider can't be null.");
 		Assert.notNull(channelProvider, "The channel provider can't be null.");
 		Assert.notNull(credentialsProvider, "The credentials provider can't be null.");
-		Assert.notNull(projectIdProvider.getProjectId(),
-				"The project ID provider can't return null.");
+		Assert.hasText(projectIdProvider.getProjectId(),
+				"The project ID provider couldn't find a configured project.");
```

Comment:
We can't make such an assumption that there was something configured at all.
The simple `project ID must not be empty` is fully enough.
With the stack trace on the matter we will see where is the problem and walking through the dependency injections we will figure out which `projectIdProvider` is guilty.